analysis/quantifier-comparisons/12-spatial-transcriptomics-storage.Rmd

```diff
@@ -0,0 +1,221 @@
+
+# read in image data using Seurat and subset to common cells
+image_data <- Seurat::Read10X_Image(spatial_dir)
+rownames(image_data@coordinates) <- gsub("-1","", rownames(image_data@coordinates))
+common_cells <- intersect(Seurat::Cells(x = fry_seurat), rownames(image_data@coordinates))
+image_data <- image_data[common_cells,]
+
+# add image data to Seurat
+Seurat::DefaultAssay(object = image_data) <- "Spatial"
+fry_seurat[['slice']] <- image_data
+```
+
+
+### Comparing Seurat and Spatial Experiment
+
+Now let's take a look at how each of these objects stores information for the experiment. 
+
+```{r}
+head(colData(fry_spe))
+head(fry_seurat@meta.data)
+```
+
+Seurat automatically adds some per cell QC metrics here, but in general the `colData` looks similar to what we would expect for a `SingleCellExperiment` for the `SpatialExperiment`.
+
+```{r}
+head(rowData(fry_spe))
+head(fry_seurat[["Spatial"]]@meta.features)
+```
+
+It looks like both of these start off with no `rowData` as expected.
+
+```{r}
+fry_spe <- fry_spe %>%
+  scuttle::addPerCellQCMetrics(subsets = list(mito_genes = mito_genes[mito_genes %in% rownames(fry_spe)])) 
+head(colData(fry_spe))
+```
+
+You can still use the same functions to add in per cell metrics that we are using in `scpcaTools` now on a `SpatialExperiment`. 
+
+Let's specifically look at how each of them stores the data related to the spatial information.
+The `SpatialExperiment` has three places where it stores spatial information, `spatialData`, `spatialCoords`, and `imgData`. 
+
+```{r}
+head(spatialData(fry_spe))
+```
+
+```{r}
+head(spatialCoords(fry_spe))
+```
+
+`spatialData` holds information about whether or not each barcode was covered by tissue and spatial metadata.
+`spatialCoords` holds the coordinates for each spot.
+
+This same information can be found by looking in the `images` slot in the `Seurat` object. 
+
+```{r}
+head(fry_seurat@images)
+```
+
+`imgData` holds information about each image that is stored as multiple slices of an image can be stored within one `SpatialExperiment`. 
+
+```{r}
+imgData(fry_spe)
+```
+
+This same information can be found by looking at the `images` slot in the `Seurat` object. 
+
+```{r}
+fry_seurat@images$slice@scale.factors
+```
+
+Let's take a look at some example plots that can be produced with each of these objects.
+
+```{r}
+library(viridis)
+# plot tissue spots coloring by total UMI count
+ggspavis::plotSpots(spe = fry_spe, 
+          x_coord = "pxl_col_in_fullres", 
+          y_coord = "pxl_row_in_fullres", 
+          annotate = "sum") +
+  scale_color_viridis_c()
+```
+You can also overlay the image by using `plotVisium`. 
+
+```{r}
+ggspavis::plotVisium(spe = fry_spe[,spatialData(fry_spe)$in_tissue == 1],
+                     x_coord = "pxl_col_in_fullres",
+                     y_coord = "pxl_row_in_fullres")
+```
+
+```{r}
+Seurat::SpatialFeaturePlot(fry_seurat, features = "nCount_Spatial")
+```
```

Comment:
If I am understanding correctly, this chunk can not run with the rest of the notebook, so we should probably make it also not evaluate. I am a strong believer that a notebook should be able to run from top to bottom without errors. 

Because of this, we might want to separately include this image as an example, with the code as present here, but the note that it was not generated from this data set due to the incompatibility.

~~~suggestion
```{r eval=FALSE}
Seurat::SpatialFeaturePlot(fry_seurat, features = "nCount_Spatial")
```
~~~